Core/src/IfcGeometryConverter/PlacementConverter.h

```diff
@@ -35,1657 +35,1664 @@
+
+                    void checkLinearPlacementAgainstAbsolutePlacement(carve::math::Matrix& matrix, std::shared_ptr<typename IfcEntityTypesT::IfcLinearPlacement> linear_placement)
+                    {
+                        // ***********************************************************
+                        // check with the provided CartesianPosition
+                        // CartesianPosition OPTIONAL
+                        carve::math::Matrix absolute_placement(carve::math::Matrix::IDENT());
+                        if(linear_placement->CartesianPosition) {
+                            convertIfcAxis2Placement3D(linear_placement->CartesianPosition.get().lock(), absolute_placement);
+                            if(absolute_placement != matrix) {
+                                BLUE_LOG(trace) << linear_placement->getErrorLog() << "Absolute placemet and the calculated linear placement do not agree";
+                            }
+                        }
+                    }
+
+                    carve::geom::vector<3> getOffsetFromCurve(double length_factor, std::shared_ptr<typename IfcEntityTypesT::IfcDistanceExpression> distExpr)
+                    {
+                        return carve::geom::VECTOR(
+                            distExpr->OffsetLongitudinal.value_or(0.0) * length_factor,
+                            distExpr->OffsetLateral.value_or(0.0) * length_factor,
+                            distExpr->OffsetVertical.value_or(0.0) * length_factor
+                        );
+                    }
+
+                    std::tuple< carve::geom::vector<3>, carve::geom::vector<3>> calculatePositionOnAndDirectionOfBaseCurve(std::shared_ptr<typename IfcEntityTypesT::IfcLinearPlacement> linear_placement, double length_factor)
+                    {
+                        carve::geom::vector<3> pointOnCurve = carve::geom::VECTOR(0.0, 0.0, 0.0);
+                        carve::geom::vector<3> directionOfCurve = carve::geom::VECTOR(1.0, 0.0, 0.0);
+                        convertBoundedCurveDistAlongToPoint3D(
+                            std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcBoundedCurve>(GetCurveOfPlacement(linear_placement)),
+                            linear_placement->Distance->DistanceAlong * length_factor,
+                            linear_placement->Distance->AlongHorizontal,
+                            pointOnCurve,
+                            directionOfCurve
+                        );
+                        return { pointOnCurve, directionOfCurve };
+                    }
+
+                    void computeRotationMatrix(carve::math::Matrix& object_placement_matrix, std::shared_ptr<typename IfcEntityTypesT::IfcLinearPlacement> linear_placement, carve::geom::vector<3> translate, carve::geom::vector<3> local_x, carve::geom::vector<3> local_y, carve::geom::vector<3> local_z)

```

Comment:
This kind of function would be beneficial for LocalPlacement as well.

Reply:
Changed out param and refactored method further